### PR TITLE
Introduce cupy.is_available()

### PR DESCRIPTION
**Pull request by niboshi**
Currently, `chainer.cuda.available` is `True` even if there is no available CUDA device (by setting `CUDA_VISIBLE_DEVICES=-1`).
In order to avoid that, this PR introduces ~~`cupy.available`~~ `cupy.is_available()` which takes account of device availability.

(Corresponding issue in PyTorch: https://github.com/pytorch/pytorch/issues/306)

### CODE REVIEW

File: tests/cupy_tests/test_init.py [hunk 1]
```diff
@@ -1,14 +1,83 @@
 import nose
+import os
+import subprocess
+import shutil
```
Comment (okuta) [inline tests/cupy_tests/test_init.py]: Please order import.

**Comment by niboshi** [comment]
Fixed.

**Comment by niboshi** [comment]
It seems calling `runtime.getDeviceCount()` in the module-level code of `cupy/cuda/__init__.py` causes problem in multi-process environment.
I fixed it by deferring the call with a function (`cupy.is_available()`) insted of a flag (`cupy.available`).

**Comment by niboshi** [comment]
Perhaps `chainer.cuda.available` should also be deferred. That would be a compatibility-breaking change.

File: cupy/__init__.py [hunk 2]
```diff
@@ -19,6 +19,14 @@
 
     six.reraise(ImportError, ImportError(msg), exc_info[2])
 
+
+from cupy import cuda
+
+
+def is_available():
+    return cuda.is_available()
```
Comment (okuta) [inline cupy/__init__.py]: How about caching this result.

Comment (niboshi) [inline cupy/__init__.py]: Fixed.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**